kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -93,8 +99,7 @@ public String sniHostname() {
      * @param header The header
      * @param message The message
      */
-    @Override
-    public void forwardRequest(RequestHeaderData header, ApiMessage message) {
+    protected void forwardRequestInternal(RequestHeaderData header, ApiMessage message) {
```

Comment:
I'm proposing to leave the `forward*()` methods on this class for this 'API' PR.   I think the next piece of work where we deal with queuing/toggling I/O help us see how we want to structure this code.  I think a hasty rehash now would just make merging with the work Rob has already done harder.